src/index.ts

```diff
@@ -54,16 +54,18 @@ export type Product = {
   position: number | string
   coupon: number | string
 }
-const eventMappings: { [key: string]: string } = {
-  pageview: 'pagevisit',
-  lead: 'lead',
-  signup: 'signup',
-  watchvideo: 'watchvideo',
-  viewcategory: 'viewcategory',
-  custom: 'custom',
-  'Product Added': 'addtocart',
-  'Order Completed': 'checkout',
-  'Products Searched': 'search',
+
+function mapEcommerceEvent(eventName: string | undefined): string | undefined {
+  switch (eventName) {
+    case 'Product Added':
+      return 'addtocart'
+    case 'Order Completed':
+      return 'checkout'
+    case 'Products Searched':
+      return 'search'
+    default:
+      return undefined // Return undefined to indicate no match
+  }
```

Comment:
What prompted this change from an object to a function with switch/case? I generally think objects are better in such a case, because you can inspect them, list all the keys, etc

Reply:
Made the change, thank you!